docker/launch-hostagent.sh

```diff
@@ -37,6 +37,75 @@ if [ -w ${PREFIX} ]; then
     mkdir -p ${VARDIR}/lib/aci-containers/k8s-pod-network
 fi
 
+function check_eth {
+    ip link show "$1" | grep -q "$1"
+}
+
+function get_ip {
+    ip addr show "$1" | grep -o 'inet [0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+' | grep -o [0-9].*
+}
+
+function get_mac {
+    ip link show "$1" | awk '/ether/ {print $2}'
+}
+
```

Comment:
L52-108 should be done conditionally in the case of overlay mode.

Reply:
This is intentional per @mandeepdhami's ask. The benefit is that we can perform health checks locally.

Reply:
Good point. We should take it from the config.